ts/client/database/stateEngine.ts

```diff
@@ -10,4 +11,16 @@ export class StateDatabaseEngine extends DatabaseEngine<State>
     constructor(depth: number) {
         super(depth, StateLeafFactory());
     }
+
+    public async updateBatch(
+        path: number,
+        depth: number,
+        items: State[]
+    ): Promise<void> {
+        for (const [i, item] of items.entries()) {
+            const itemID = path * 2 ** depth + i;
+            await this.update(itemID, item);
+            await Pubkey2StatesDB.update(item.pubkeyID.toNumber(), itemID);
+        }
+    }
```

Comment:
overriding `updateBatch` method to update `pubkey->stateID` mapping when new state is created